src/QMCWaveFunctions/Fermion/MultiDiracDeterminant_help.h

```diff
@@ -24,84 +23,136 @@
+        a21 *
+            (a12 * (a33 * (a44 * a55 - a54 * a45) - a43 * (a34 * a55 - a54 * a35) + a53 * (a34 * a45 - a44 * a35)) -
+             a32 * (a13 * (a44 * a55 - a54 * a45) - a43 * (a14 * a55 - a54 * a15) + a53 * (a14 * a45 - a44 * a15)) +
+             a42 * (a13 * (a34 * a55 - a54 * a35) - a33 * (a14 * a55 - a54 * a15) + a53 * (a14 * a35 - a34 * a15)) -
+             a52 * (a13 * (a34 * a45 - a44 * a35) - a33 * (a14 * a45 - a44 * a15) + a43 * (a14 * a35 - a34 * a15))) +
+        a31 *
+            (a12 * (a23 * (a44 * a55 - a54 * a45) - a43 * (a24 * a55 - a54 * a25) + a53 * (a24 * a45 - a44 * a25)) -
+             a22 * (a13 * (a44 * a55 - a54 * a45) - a43 * (a14 * a55 - a54 * a15) + a53 * (a14 * a45 - a44 * a15)) +
+             a42 * (a13 * (a24 * a55 - a54 * a25) - a23 * (a14 * a55 - a54 * a15) + a53 * (a14 * a25 - a24 * a15)) -
+             a52 * (a13 * (a24 * a45 - a44 * a25) - a23 * (a14 * a45 - a44 * a15) + a43 * (a14 * a25 - a24 * a15))) -
+        a41 *
+            (a12 * (a23 * (a34 * a55 - a54 * a35) - a33 * (a24 * a55 - a54 * a25) + a53 * (a24 * a35 - a34 * a25)) -
+             a22 * (a13 * (a34 * a55 - a54 * a35) - a33 * (a14 * a55 - a54 * a15) + a53 * (a14 * a35 - a34 * a15)) +
+             a32 * (a13 * (a24 * a55 - a54 * a25) - a23 * (a14 * a55 - a54 * a15) + a53 * (a14 * a25 - a24 * a15)) -
+             a52 * (a13 * (a24 * a35 - a34 * a25) - a23 * (a14 * a35 - a34 * a15) + a33 * (a14 * a25 - a24 * a15))) +
+        a51 *
+            (a12 * (a23 * (a34 * a45 - a44 * a35) - a33 * (a24 * a45 - a44 * a25) + a43 * (a24 * a35 - a34 * a25)) -
+             a22 * (a13 * (a34 * a45 - a44 * a35) - a33 * (a14 * a45 - a44 * a15) + a43 * (a14 * a35 - a34 * a15)) +
+             a32 * (a13 * (a24 * a45 - a44 * a25) - a23 * (a14 * a45 - a44 * a15) + a43 * (a14 * a25 - a24 * a15)) -
+             a42 * (a13 * (a24 * a35 - a34 * a25) - a23 * (a14 * a35 - a34 * a15) + a33 * (a14 * a25 - a24 * a15))));
   }
 
+  /** default implementation of MultiDiracDeterminant::CalculateRatioFromMatrixElements
+   *  If you don't have a perfect square below 25 of dots this is what you hit
+   *  dots must be a 2n by 2n Matrix
+   *  iter must be size n^2
+   *  this object must have been resized to n
+   */
   template<typename ITER>
   inline T evaluate(Matrix<T>& dots, ITER it, int n)
   {
-    T* restrict d = M.data();
-    for(int i=0; i<n; i++)
-      for(int j=0; j<n; j++)
-        *(d++)= dots(*(it+i),*(it+n+j));
-    return Determinant(d,n,n,Pivot.data());
-  }
+#ifndef NDEBUG
+    //Do bounds checking
+    if (static_cast<size_t>(n ^ 2) > M.size())
+    {
+      std::string abort_message("MultiDiracDeterminantCalculator::evaluate(Matrix<T>& dots, ITER it, int n)");
+      abort_message += " failed, n^2 (" + std::to_string(n ^ 2) + ") greater than M.size() (";
+      abort_message += std::to_string(M.size()) + "\n";
+      APP_ABORT(abort_message);
```

Comment:
#318 describe the conflict between APP_ABORT and unit tests.
Since the added unit test covers this class, it is better to avoid using APP_ABORT.
Another reason is that the underneath MPI_Abort can be a potential problem.
I also feel is not a good idea to use NDEBUG which is supposed for C assert instead of us.
Could you convert the whole error message handing to a single assert()

Reply:
I was on the fence about this anyway.  I'll just drop the bounds check.